haystack/schema.py

```diff
@@ -854,7 +867,10 @@ def _build_answer_metrics_df(self,
         - f1 (How well does the best matching returned results overlap with any gold answer on token basis?)
         - sas if a SAS model has bin provided during during pipeline.eval() (How semantically similar is the prediction to the gold answers?)
         """
-        queries = answers["query"].unique()
+        # sort gold_document_ids and convert to tuple to make them hashable
+        answers["gold_document_ids"] = answers["gold_document_ids"].apply(lambda x: tuple(sorted(x)))
+        answers.insert(loc=0, column='query_id', value=answers.set_index(['query', 'gold_document_ids']).index.factorize()[0] + 1)
+        answers["gold_document_ids"] = answers["gold_document_ids"].apply(lambda x: list(x))
```

Comment:
Wouldn't it be better understandable if we made query_id a property of MultiLabel? That way we could set it to the eval dataframes too, not only the metrics dataframes here. That would it make easier for analyzing the data.

Reply:
Yes, done. Let's discuss the naming though. Currently, it's `query_id`, which could be misleading because the same query text can correspond to different query ids if they are in labels with different filters.